publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java

```diff
@@ -947,7 +950,30 @@ void shouldMaintainAssociatedArtifactsThatOtherThanFile() throws ApiGatewayExcep
         assertThat(persistedPublished.getStatus(), is(equalTo(PUBLISHED)));
         assertThat(persistedPublished.getAssociatedArtifacts(), everyItem(is(instanceOf(AssociatedLink.class))));
     }
-    
+
+    @Test
+    void shouldBePossibleToSetPublicationStatusToDeletedForPublishedPublication() throws ApiGatewayException {
+        var publishedResource = createPublishedResource();
+        var publicationIdentifier = publishedResource.getIdentifier();
+        var expectedUpdateStatus = UpdateResourceService.deletionStatusChangeInProgress();
+        var actualUpdateStatus = resourceService.updatePublishedStatusToDeleted(publicationIdentifier);
+        assertThat(actualUpdateStatus, is(equalTo(expectedUpdateStatus)));
+        var actualPublicationInDatabaseAfterStatusUpdate =
+            resourceService.getPublicationByIdentifier(publicationIdentifier);
+        assertThat(actualPublicationInDatabaseAfterStatusUpdate.getStatus(), is(equalTo(PublicationStatus.DELETED)));
+        assertThat(actualPublicationInDatabaseAfterStatusUpdate.getPublishedDate(), is(equalTo(null)));
+    }
+
+    @Test
+    void updatePublishedStatusToDeletedShouldReturnResourceAlreadyDeletedMessage() throws ApiGatewayException {
+        var publishedResource = createPublishedResource();
+        var publicationIdentifier = publishedResource.getIdentifier();
+        var expectedUpdateStatus = UpdateResourceService.deletionStatusIsCompleted();
+        resourceService.updatePublishedStatusToDeleted(publicationIdentifier);
+        var actualUpdateStatus = resourceService.updatePublishedStatusToDeleted(publicationIdentifier);
```

Comment:
The two identical calls to updatePublishedStatusToDeleted are a bit weird, but I didn't manage to insert a mock deleted publication into the database without the resourceService changing it to "DRAFT".